x-pack/plugin/core/src/test/java/org/elasticsearch/snapshots/SourceOnlySnapshotShardTests.java

```diff
@@ -236,7 +235,6 @@ public void testRestoreMinmal() throws IOException {
         IndexMetadata metadata = runAsSnapshot(threadPool, () -> repository.getSnapshotIndexMetadata(snapshotId, indexId));
         IndexShard restoredShard = newShard(
             shardRouting, metadata, null, SourceOnlySnapshotRepository.getEngineFactory(), () -> {}, RetentionLeaseSyncer.EMPTY);
-        restoredShard.mapperService().merge(shard.indexSettings().getIndexMetadata(), MapperService.MergeReason.MAPPING_RECOVERY);
```

Comment:
I'll double check with someone from the distributed team that this change is okay.

Reply:
@original-brownbear would you be up for taking a look? I saw that you made several changes in this area recently.

Reply:
I think this is fine. This line seems unnecessary and makes the test behave less like a real restore as far as I can see anyway (just restoring the meta from the repo in a real restore).
=> LGTM :)

Reply:
Thanks @original-brownbear ! It also seemed strange to me, that we would add the original mappings as part of restoring a source-only snapshot.